src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -109,6 +115,18 @@ private void EndRequest(HttpContext httpContext, long timestamp)
                     telemetry.Success &= successExitCode;
                 }
 
+                IHeaderDictionary requestHeaders = httpContext.Request?.Headers;
```

Comment:
Some clarifications: we changed the implementation of this logic in classic ASP.NET to allow customer to override this value. As minimum - check that this value is not set yet by telemetry initializers. Ideally - move this logic to `OperationCorrelationTelemetryInitializer`